htdocs/core/class/commondocgenerator.class.php

```diff
@@ -587,7 +587,25 @@ public function get_substitutionarray_lines($line, $outputlangs)
 		$extralabels = $extrafields->fetch_name_optionals_label($extrafieldkey, true);
 		$line->fetch_optionals();
 
-		$resarray = $this->fill_substitutionarray_with_extrafields($line, $resarray, $extrafields, $array_key, $outputlangs);
+        $resarray = $this->fill_substitutionarray_with_extrafields($line, $resarray, $extrafields, $array_key, $outputlangs);
+        
+        // Check if the current line belongs to a supplier order
+        if (get_class($line) == 'CommandeFournisseurLigne') {
+            // Add the product supplier extrafields to the substitutions
+            $extralabels=$extrafields->fetch_name_optionals_label("product_fournisseur_price");
+            $columns = "";
+            foreach ($extralabels as $key => $value) {
```

Comment:
$extralabels is deprecated. 
You must use   $extrafields->attributes["product_fournisseur_price"]['label']    instead.

Reply:
It would be nice if this would also be mentioned here in the Wiki: https://wiki.dolibarr.org/index.php/Extrafields#Load_extrafield_array_in_object